packages/api/portfolios/environments/createEnvironment.ts

```diff
@@ -0,0 +1,109 @@
+import "reflect-metadata";
+import { createConnection } from "../../utils/database";
+import { Portfolio } from "../../../orm/entity/Portfolio";
+import { Application } from "../../../orm/entity/Application";
+import { Environment, IEnvironment, IEnvironmentCreate } from "../../../orm/entity/Environment";
+import { EnvironmentRepository } from "../../repository/EnvironmentRepository";
+import { APIGatewayProxyResult, Context } from "aws-lambda";
+import { ApiSuccessResponse, SuccessStatusCode } from "../../utils/response";
+import internalSchema = require("../../models/internalSchema.json");
+import middy from "@middy/core";
+import xssSanitizer from "../../portfolioDrafts/xssSanitizer";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import validator from "@middy/validator";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import cors from "@middy/http-cors";
+import { ApiGatewayEventParsed } from "../../utils/eventHandlingTool";
+import { CORS_CONFIGURATION } from "../../utils/corsConfig";
+import { wrapSchema } from "../../utils/schemaWrapper";
+import { errorHandlingMiddleware } from "../../utils/errorHandlingMiddleware";
+import createError from "http-errors";
+import { IpCheckerMiddleware } from "../../utils/ipLogging";
+import { validateRequestShape } from "../../utils/shapeValidator";
+
+/**
+ * Submits the environment of an application
+ *
+ * @param event - The POST request from API Gateway
+ */
+export async function baseHandler(
+  event: ApiGatewayEventParsed<IEnvironment>,
+  context?: Context
+): Promise<APIGatewayProxyResult> {
+  const setupResult = validateRequestShape<IEnvironment>(event);
+  const { portfolioId, applicationId } = setupResult.path;
+  const environmentBody = setupResult.bodyObject;
```

Comment:
You don't need to set this, `.use(jsonBodyParser())` parses the event.body before anything inside the baseHandler actually runs

Reply:
I'm not sure I completely follow, should I just use `event.body`? Also, do we not want to use what is coming from `validateRequestShape`?

Reply:
Understood from our conversation earlier. I left as-is to avoid typescript issues.